puppet/ha-puppet/screenshot.js

```diff
@@ -184,7 +191,29 @@ export class Browser {
     return this.page;
   }
 
-  async screenshotHomeAssistant({ pagePath, viewport, extraWait }) {
+  async _updateTheme(darkMode) {
+    await this.page.evaluate((darkMode) => {
+      localStorage.setItem("selectedTheme", `{"dark": ${darkMode === true}}`);
+      const event = new CustomEvent("settheme", {
+        bubbles: true,
+        composed: true,
+        detail: { dark: darkMode === true }
+      });
+      const homeAssistant = document.querySelector("home-assistant");
+      if (homeAssistant) {
+        homeAssistant.dispatchEvent(event);
+      } else {
+        window.dispatchEvent(event);
+      }
+    }, darkMode);
+
+    this.currentDarkMode = darkMode;
+
+    // Give the theme a moment to apply
+    await new Promise(resolve => setTimeout(resolve, 300));
```

Comment:
Instead of doing this in getPage, we should do this inside the screenshot function. Since we already have a defaultWait, we can just make sure that it is at least 300ms when we have changed the theme. We don't want to add the wait for both theme and navigation change together.